code/modules/mob/death.dm

```diff
@@ -56,8 +56,9 @@
 	if(transmogged_from)
 		transmog_death()
 	if(client || mind)
+		var/mobname = src.real_name ? "[src.real_name]" : "[src]"
 		for(var/mob/M in get_deadchat_hearers())
-			var/rendered = "\proper<a href='?src=\ref[M];follow2=\ref[M];follow=\ref[src]'>(Follow)</a><span class='game deadsay'> \The <span class='name'>[src]</span> has died at \the <span class='name'>[get_area(src)]</span>.</span>"
+			var/rendered = "\proper<a href='?src=\ref[M];follow2=\ref[M];follow=\ref[src]'>(Follow)</a><span class='game deadsay'> \The <span class='name'>[mobname]</span> has died at \the <span class='name'>[get_area(src)]</span>.</span>"
```

Comment:
how about 
```suggestion
			var/rendered = "\proper<a href='?src=\ref[M];follow2=\ref[M];follow=\ref[src]'>(Follow)</a><span class='game deadsay'> \The <span class='name'>[mobname] (as [src])</span> has died at \the <span class='name'>[get_area(src)]</span>.</span>"
```
?

Reply:
I'd only want that to display if the name they died as doesn't match their real name. I'll account for this, one moment.

Reply:
Take a look now. I more or less copied how deadchat does it, so it should work fine:
https://github.com/vgstation-coders/vgstation13/blob/0560e402fd0a1ce0d3d2bf340ca95902e59afe10/code/modules/mob/say.dm#L50-L67